core/generator.js

```diff
@@ -275,6 +275,8 @@ Blockly.Generator.prototype.valueToCode = function(block, name, outerOrder) {
 
 /**
  * Generate code representing the statement.  Indent the code.
+ * This is mainly used in generators. When trying to generate code to evaluate
```

Comment:
In addition, change "Generate code representing the statement." to "Generate a code string representing the blocks attached to the named statement input."

Between the two changes, hopefully we'll reduce confusion.